core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java

```diff
@@ -253,7 +258,17 @@ private void updateStates(Stack stack, Collection<InstanceMetaData> failedInstan
         clusterService.updateClusterCertExpirationState(stack.getCluster(), hostCertExpiring);
         clusterOperationService.reportHealthChange(stack.getResourceCrn(), newFailedNodeNamesWithReason, newHealthyHostNames);
         if (!failedInstances.isEmpty()) {
-            clusterService.updateClusterStatusByStackId(stack.getId(), DetailedStackStatus.NODE_FAILURE);
+            if (stackUtil.stopStartScalingEnabled(stack)) {
```

Comment:
There is only one scenario where it could happen that node is running but the CM status shows in maintenance mode, that is, when the start cluster fails to recommission the nodes. It will be in start failed state and the status sync job will set it to available. Need to check on CM based status syncer job for this, if there is one.

Reply:
From what I could see, CM status is taken into account, but the node in maintenance mode is not taken into account. Added a comment for the same.

Reply:
Found another scenario, when a customer repairs the stopped compute node, it ends up in running state in cloud provider and with decommissioned state in CM.

Reply:
I can address this as part of CB-15150.